### PR TITLE
blockchain: test vectors for proto encoding

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

this PR adds test vectors for proto encoding. the main difference from amino was the removal of four bytes due to interface encoding.

should i add more cases?

Closes: #XXX

### CODE REVIEW

File: blockchain/msgs_test.go [hunk 1]
```diff
@@ -86,3 +90,50 @@ func TestBcStatusResponseMessageValidateBasic(t *testing.T) {
...
+
+	testCases := []struct {
+		testName string
+		bmsg     proto.Message
+		expBytes []byte
+	}{
+		{"BlockRequestMessage", &bcproto.BlockRequest{Height: 1}, []byte{0x8, 0x1}},
+		{"BlockRequestMessage", &bcproto.BlockRequest{Height: math.MaxInt64},
+			[]byte{0x8, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f}},
+		{"BlockResponseMessage", &bcproto.BlockResponse{Block: bpb}, []byte{0xa, 0x6c, 0xa, 0x37, 0xa, 0x0, 0x18,
```
Comment (melekes) [inline blockchain/msgs_test.go:109]: how one can generate these bytes?

Comment (marbar3778) [reply blockchain/msgs_test.go:109]: the ones in `[]byte{..}`? this is the output of using `require.Equal` and it failing. `fmt.Println(..)` outputs it into numbers. This is to keep consistent with how it is done in vote test vectors.

**Comment by marbar3778** [comment]
> Btw, I usually find it easier to express the vectors as hex strings, since the diffs are more readable and it's easier to copy/paste them.

I can make this change, I was mainly keeping this format to be consistent with other test vectors

**Comment by erikgrinaker** [comment]
> > Btw, I usually find it easier to express the vectors as hex strings, since the diffs are more readable and it's easier to copy/paste them.
> 
> 
> 
> I can make this change, I was mainly keeping this format to be consistent with other test vectors

Nah, just a tip - either format works

**Pull request merged**